blog/content/post/2024-08-19-secureboot-freebsd.md

```diff
@@ -0,0 +1,378 @@
+
+1. Select `Reset to default Secure Boot Keys`
+    ![reset to default sboot keys](/img/reset_to_def.png)
+
+1. Select `YES`
+
+1. Press `F10` to save and `Y` to confirm
+    ![f10 save dialog](/img/save_conf.png)
+
+1. Reset the platform by pressing `ESC` twice and selecting `Reset`
+
+#### Enrolling the ceritficate
+
+1. Press `F2` to enter setup again
+    ![setup menu](/img/setup_root.png)
+
+1. Navigate to `Device Manager/Secure Boot Configuration`
+    ![secure boot enabled](/img/sboot_enabled.png)
+
+1. Enable `Custom Mode` and enter the `Advanced Secure Boot Keys Management` 
+    menu again.
+    ![advanced secure boot keys management](/img/advanced_sboot.png)
+
+1. Navigate to `DB Options/Enroll Signature/Enroll Signature Using File/`.
+    If the EFI directory is properly mounted, you should see a single entry in
+    this menu, similar to this
+     ![efi directory entry](/img/efi_entry.png)
+
+1. Select the entry, and then select `testcert.cer`
+     ![testcert.cer](/img/testcert.png)
+
+1. Select `Commit Changes and Exit`
+     ![commit changes and exit](/img/commit_changes.png)
+
+1. Press `F10` to save and `Y` to confirm
+    ![f10 save dialog](/img/save_conf.png)
+
+1. Make sure that `Current Secure Boot State` is enabled
+    ![secure boot enabled](/img/sboot_enabled.png)
+
+1. Reset the platform again
+
+### Booting FreeBSD
+
+1. Wait for Dasharo to boot into UEFI Shell
+    ![EFI shell](/img/efi_shell.png)
+
+1. Enter the filesystem and boot the signed-loader.efi
+  ```
+  fs0:
+  signed-loader.efi
+  ```
+  ![EFI shell commands](/img/ush_commands.png)
+
+1. Say hello to Beastie
+    ![beastie](/img/beastie.png)
+
+
+## Testing on hardware
+
+Upon making sure that the loader-kernel object boots properly within an 
+emulation environment, we can proceed to Secure Booting FreeBSD on hardware.
+
+To do that, follow the exact same steps as with emulation, the only differewnce
+being that you will now need to upload the certificate to a USB drive and 
+enroll the certificate from there. 
+
+You will also need to place the `signed-loader.efi` file in your EFI partition,
+and add it as a custom boot option.
+
+### Adding a custom boot option
+
+* Enter setup menu
+* Navigate to `Boot Maintenance Manager/Boot Options/Add Boot Option`
+* Choose the appropriate disk label. They might look intimidating, but you
+  should be able to find the correct one by looking for a familiar keyword.
+  If you have an NVME drive for example, there should be an entry with 
+  `Pci(0x0,0x0)/NVMe`.
+* Find the loader-kernel object. It should be located under 
+  `<efi>/<freebsd>/loader.efi`
+* Name the entry appropriately, confirm and save the changes.
+
+
+## Troubleshooting
+
+### Mountroot
+
+If after booting the `signed-loader.efi` you land in `mountroot`, you need to
+type in `zfs:zroot/ROOT/default` and add the line
+
+```bash
+vfs.root.mountfrom="zfs:zroot/ROOT/default"
+```
+
+to your `/boot/loader.conf`.
+
+## Summary
+
```

Comment:
Actual technical summary could be there.

Reply:
[Added](https://github.com/3mdeb/news-and-ideas/pull/603/commits/1f14bdc84a7b4847b01bb79417f61d5d05422bf2#diff-1e142bc7519e97361fc0d3867abcb7fa3d257a4b5ccb79795e5e00acebf7d22fR376)